### PR TITLE
Add entt

### PR DESCRIPTION
**Pull request by yamashi**

### CODE REVIEW

File: packages/e/entt/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    set_homepage("https://github.com/skypjack/entt")
+    set_description("Gaming meets modern C++ - a fast and reliable entity component system (ECS) and much more.")
+
+    set_urls("https://github.com/skypjack/entt/archive/$(version).tar.gz",
+             "https://github.com/skypjack/entt.git")
+
+    add_versions("v3.6.0", "94b7dc874acd0961cfc28cf6b0342eeb0b7c58250ddde8bdc6c101e84b74c190")
+    add_deps("cmake")
+
+    on_install("*", function (package)
```
Comment (waruqi) [inline packages/e/entt/xmake.lua:12]: please remove "*"

File: packages/e/entt/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        local configs = {}
```
Comment (waruqi) [inline packages/e/entt/xmake.lua:13]: and shared/static library 

```lua
        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
```

File: packages/e/entt/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    set_homepage("https://github.com/skypjack/entt")
+    set_description("Gaming meets modern C++ - a fast and reliable entity component system (ECS) and much more.")
+
+    set_urls("https://github.com/skypjack/entt/archive/$(version).tar.gz",
+             "https://github.com/skypjack/entt.git")
+
+    add_versions("v3.6.0", "94b7dc874acd0961cfc28cf6b0342eeb0b7c58250ddde8bdc6c101e84b74c190")
+    add_deps("cmake")
+
+    on_install("", function (package)
```
Comment (waruqi) [inline packages/e/entt/xmake.lua:12]: we only need `on_install(function (package)`, if this package support all platforms

File: packages/e/entt/xmake.lua (hunk 2 above)
Comment (yamashi) [reply packages/e/entt/xmake.lua:13]: It's header only

File: packages/e/entt/xmake.lua (hunk 3 above)
Comment (waruqi) [reply packages/e/entt/xmake.lua:12]: need not `""`

File: packages/e/entt/xmake.lua (hunk 2 above)
Comment (waruqi) [reply packages/e/entt/xmake.lua:13]: ok

**Pull request merged**